### PR TITLE
Arm Subsystem

### PR DESCRIPTION
**Pull request by gxu0904**
- arm subsystem
- commands to move the arm to ground position and shooter position.

### CODE REVIEW

**Comment by ChaoticChaosTheGreat** [comment]
Wait is there a reason we used WPILIB's PIDController instead of the TalonFx one? Would be more standard + more accurate if we used the one from the phoenix 6 library. 
See motion profiling and position control here: https://pro.docs.ctr-electronics.com/en/latest/docs/api-reference/device-specific/talonfx/basic-pid-control.html#position-control

**Comment by realjoshuau** [comment]
Arm subsystem has a separate absolute encoder that doesn't allow us to use TalonFX PID.

**Comment by realjoshuau** [comment]
The TalonFX encoder is **not** absolute, so you can't reliably set positions on them.

**Comment by ChaoticChaosTheGreat** [comment]
Ah I see. So no motion magic then?

**Comment by 3LucasZ** [comment]
I think we will use PIDCommand instead of PIDSubsystem, since that's how we've been doing it

**Comment by 3LucasZ** [comment]
There's only 2 angles (arm fully up and arm fully down), we can PID all the way up/down really fast and stop when the encoder doesn't notice signficant change.

**Comment by realjoshuau** [comment]
Design has changed this subsystem **significantly**.  Look at this issue: #116, and submit another PR. You should base it off of this, since there are not many changes.

**Comment by realjoshuau** [comment]
Design has changed this subsystem **significantly** since assembly refuses to add sensors. Look at this issue: #116, and submit another PR. You should base it off of this, since there are not many changes.

File: vendordeps/REVLib-2024.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
+{
```
Comment (3LucasZ) [inline vendordeps/REVLib-2024.json:1]: What is this file for?

**Review by 3LucasZ — CHANGES_REQUESTED** [review]
Nice job! I have some small requests. Next step will be sim.

**Review by realjoshuau — APPROVED** [review]
whatever

**Pull request merged**